File: src/openrgb_flowers/effects/palettes/aurora_palette.py
"""Nordic Aurora Borealis organic color palette."""
from openrgb_flowers.effects.palettes.base_palette import BaseColorPalette
from openrgb_flowers.core.models.color_rgb import ColorRGB


class AuroraPalette(BaseColorPalette):
    """Aurora Borealis: Vivid Polar Green -> Emerald -> Turquoise -> Royal Violet."""

    def __init__(self) -> None:
        super().__init__(
            name="aurora",
            stops=[
                (0.0, ColorRGB(0, 255, 135)),    # Polar Green
                (0.3, ColorRGB(96, 239, 255)),   # Turquoise
                (0.6, ColorRGB(0, 180, 216)),    # Deep Ice Blue
                (0.85, ColorRGB(123, 44, 191)),  # Royal Violet
                (1.0, ColorRGB(224, 86, 253)),   # Radiant Magenta
            ],
        )

File: src/effects/palettes/aurora_palette.cpp
#include "openrgb_flowers/effects/palettes/aurora_palette.hpp"

namespace openrgb_flowers::effects::palettes {

AuroraPalette::AuroraPalette()
    : BaseColorPalette("aurora", {
        {0.00f, core::models::ColorRGB(0, 255, 135)},    // Polar Green
        {0.30f, core::models::ColorRGB(96, 239, 255)},   // Turquoise
        {0.60f, core::models::ColorRGB(0, 180, 216)},    // Deep Ice Blue
        {0.85f, core::models::ColorRGB(123, 44, 191)},   // Royal Violet
        {1.00f, core::models::ColorRGB(224, 86, 253)},   // Radiant Magenta
    }) {}

} // namespace openrgb_flowers::effects::palettes
